### ISSUE TITLE
dnf enablerepo not working

### ISSUE THREAD

**Issue by dustymabe** [ansible/ansible-modules-extras#885]
On fedora 22 (`dnf-1.0.0-1.fc22.noarch` and `ansible-1.9.2-1.fc22.noarch`) I can't seem to get enablerepo to work with the dnf module. I am using these lines:

```
    - name: install docker
      dnf: name=docker state=latest enablerepo='updates-testing'
```

Here is what I get when I run the playbook:

```
TASK: [install docker] ******************************************************** 
failed: [f22] => {"failed": true, "parsed": false}
Traceback (most recent call last):
  File "/root/.ansible/tmp/ansible-tmp-1440643378.23-192604758149271/dnf", line 2434, in <module>
    main()
  File "/root/.ansible/tmp/ansible-tmp-1440643378.23-192604758149271/dnf", line 824, in main
    disablerepo, disable_gpg_check)
  File "/root/.ansible/tmp/ansible-tmp-1440643378.23-192604758149271/dnf", line 749, in ensure
    my.repos.enableRepo(r)
AttributeError: 'RepoDict' object has no attribute 'enableRepo'
OpenSSH_6.9p1, OpenSSL 1.0.1k-fips 8 Jan 2015
debug1: Reading configuration data /etc/ssh/ssh_config
debug1: /etc/ssh/ssh_config line 56: Applying options for *
debug1: auto-mux: Trying existing master
debug1: mux_client_request_session: master session id: 2
Shared connection to 192.168.121.242 closed.


FATAL: all hosts have already failed -- aborting
```

Can someone else confirm this is happening for them?

**Comment by abadger** [comment]
Replicated here.

@DJMuggs do you have time to look into how to fix this?

**Comment by abadger** [comment]
Just merged a rewrite of the dnf module from some contributors.  https://github.com/ansible/ansible-modules-extras/pull/527  I tested enablerepo and disablerepo and it appears that this is fixed.  Because it is such a major change this might not be backported to the 1.9.x branch but if you need it before 2.0 is out you should be able to copy https://raw.githubusercontent.com/ansible/ansible-modules-extras/devel/packaging/os/dnf.py locally to replace the one shipped with 1.9.3

# Closing This Ticket

Hi!

We believe recent commits (likely detailed above) should resolve this question or problem for you.

This will also be included in the next major release.

If you continue seeing any problems related to this issue, or if you have any further questions, please let us know by stopping by one of the two mailing lists, as appropriate:
- https://groups.google.com/forum/#!forum/ansible-project - for user questions, tips, and tricks
- https://groups.google.com/forum/#!forum/ansible-devel - for strategy, future planning, and questions about writing code

Because this project is very active, we're unlikely to see comments made on closed tickets, but the mailing list is a great way to ask questions, or post if you don't think this particular
issue is resolved.

Thank you!
